src/InstantiableEntity.php

```diff
@@ -8,8 +8,12 @@
 
 class InstantiableEntity extends BaseEntity {
 
-	function __construct( $name, array $properties = [] ) {
-		$this->name = $name;
+	protected $name;
+
+	function __construct( $name = null, array $properties = [] ) {
+		if ( isset( $name ) ) {
```

Comment:
The check is the whole point of this pull request - `$name` may be set in a derivative class, in which case it does not need to be passed in the constructor, and, more importantly, it cannot be passed when using the `::instance()` method.